plugin/https/README.md

```diff
@@ -0,0 +1,103 @@
+# https
+
+## Name
+
+*https* - facilitates proxying DNS messages to upstream resolvers via DoH protocol.
+
+## Description
+
+The *https* plugin supports DoH.
+
+This plugin can only be used once per Server Block.
+
+## Syntax
+
+In its most basic form:
+
+~~~
+https FROM TO...
```

Comment:
this form is probably never valid because you need to set the `tls_servername`...

Reply:
It is not always necessary to specify `tls_servername`. This Corefile works

```
. {
    bind 127.0.0.1
    https . dns.quad9.net/dns-query {
        policy round_robin
    }
}
```

Reply:
you can't connect to a dns name, you can't guarantee that you can resolve it. *And* it needs re-resolving every so often

Reply:
RFC 8484 Section 10 explicitly states that:
> A DoH client may face a similar bootstrapping problem when the HTTP
   request needs to resolve the hostname portion of the DNS URI.  Just
   as the address of a traditional DNS nameserver cannot be originally
   determined from that same server, a DoH client cannot use its DoH
   server to initially resolve the server's host name into an address.
   Alternative strategies a client might employ include 1) making the
   initial resolution part of the configuration, 2) IP-based URIs and
   corresponding IP-based certificates for HTTPS, or 3) resolving the
   DNS API server's hostname via traditional DNS or another DoH server
   while still authenticating the resulting connection via HTTPS.

Your case is point 3 and it already works.

> And it needs re-resolving every so often

Almost all DoH upstream servers support HTTP2 protocol and its support is adviced by RFC 8484 Section 5.2. And I enforced its usage [here](https://github.com/coredns/coredns/pull/4583/commits/7e35bec317c11ce7944d88d1f8bdee70567bc3a1#diff-6b083e7eb24915559d0eb45d51e3f57aa5273afa9e2a34727517ce25a8d1280dR38). So DNS resolving will obviously not happen for every DNS request.

> you can't guarantee that you can resolve it

In this case user can use long form syntax:

```
. {
    bind 127.0.0.1
    https . 9.9.9.9:5053/dns-query {
      tls_servername quad9.net
    }
}
```

you can try it yourself and see if it works.

Reply:
@miekg can this conversation be resolved now ?